src/Catalyst.Core.Modules.Keystore/KeyStoreService.cs

```diff
@@ -0,0 +1,830 @@
+            if (key == null)
+            {
+                return null;
+            }
+
+            return new KeyInfo
+            {
+                Id = key.Id, Name = key.Name
+            };
+        }
+
+        /// <summary>
+        ///     Gets the IPFS encoded public key for the specified key.
+        /// </summary>
+        /// <param name="name">
+        ///     The local name of the key.
+        /// </param>
+        /// <param name="cancel">
+        ///     Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException" /> is raised.
+        /// </param>
+        /// <returns>
+        ///     A task that represents the asynchronous operation. The task's result is
+        ///     the IPFS encoded public key.
+        /// </returns>
+        /// <remarks>
+        ///     The IPFS public key is the base-64 encoding of a protobuf encoding containing
+        ///     a type and the DER encoding of the PKCS Subject Public Key Info.
+        /// </remarks>
+        /// <seealso href="https://tools.ietf.org/html/rfc5280#section-4.1.2.7" />
+        public async Task<string> GetPublicKeyAsync(string name, CancellationToken cancel = default)
+        {
+            // TODO: Rename to GetIpfsPublicKeyAsync
+            string result = null;
+            var ekey = await Store.TryGetAsync(name, cancel).ConfigureAwait(false);
+            if (ekey != null)
+            {
+                UseEncryptedKey(ekey, key =>
+                {
+                    var kp = GetKeyPairFromPrivateKey(key);
+                    var spki = SubjectPublicKeyInfoFactory.CreateSubjectPublicKeyInfo(kp.Public).GetDerEncoded();
+
+                    // Add protobuf cruft.
+                    var publicKey = new PublicKey
+                    {
+                        Data = spki,
+                        Type = kp.Public switch
+                        {
+                            RsaKeyParameters _ => KeyType.Rsa,
+                            Ed25519PublicKeyParameters _ => KeyType.Ed25519,
+                            ECPublicKeyParameters _ => KeyType.Secp256K1,
+                            _ => throw new NotSupportedException(
+                                $"The key type {kp.Public.GetType().Name} is not supported.")
+                        }
+                    };
+
+                    using var ms = new MemoryStream();
+                    Serializer.Serialize(ms, publicKey);
+                    result = Convert.ToBase64String(ms.ToArray());
+                });
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public async Task<IKey> CreateAsync(string name,
+            string keyType,
+            int size,
+            CancellationToken cancel = default)
+        {
+            // Apply defaults.
+            if (string.IsNullOrWhiteSpace(keyType))
+            {
+                keyType = Options.DefaultKeyType;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'keyType'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)